src/pt/manhastro/src/eu/kanade/tachiyomi/extension/pt/manhastro/Manhastro.kt

```diff
@@ -1,345 +1,168 @@
-            .add("nonce", nonce)
-            .build()
-
-        val response = defaultClient.newCall(
-            POST(
-                "$baseUrl/wp-admin/admin-ajax.php",
-                formHeaders,
-                form,
-                CacheControl.FORCE_NETWORK,
-            ),
-        ).execute()
-        val message = """
-            Falha ao acessar recurso: Usuário ou senha incorreto.
-            Altere suas credências em Extensões > $name > Configuração.
-        """.trimIndent()
-
-        response.use {
-            if (it.isSuccessful.not()) {
-                showToast(message)
-            }
-        }
+    override fun getChapterUrl(chapter: SChapter): String {
+        val ids = chapter.url.split(":")
+        return "$baseUrl/$mangaSubString/${ids.first()}/chapter/${ids.last()}"
+    }
 
-        return response.headers("Set-Cookie")
-            .firstOrNull { it.contains("wordpress_logged_in_", ignoreCase = true) }
-            ?.let(::Cookie)
-            ?: Cookie.empty().also {
-                showToast(message)
-            }
+    override fun chapterListParse(response: Response): List<SChapter> {
+        val mangaId = response.request.url.pathSegments.last().toInt()
+        return response.parseAs<ResponseWrapper<List<ChapterDto>>>().data
+            .map { it.toSChapter(mangaId) }
+            .sortedByDescending(SChapter::chapter_number)
     }
 
-    // ============================ Utilities ====================================
+    // ============================== Pages =====================================
 
-    private fun OkHttpClient.Builder.addNetworkInterceptorIf(
-        condition: Boolean,
-        interceptor: Interceptor,
-    ): OkHttpClient.Builder {
-        if (condition) {
-            addNetworkInterceptor(interceptor)
-        }
-        return this
+    override fun pageListRequest(chapter: SChapter): Request {
+        val chapterId = chapter.url.substringAfterLast(":")
+        return GET("$apiUrl/paginas/$chapterId", headers)
+    }
+
+    override fun pageListParse(response: Response): List<Page> {
+        return response.parseAs<ResponseWrapper<PageDto>>().data.toPageList()
     }
 
-    private val handler by lazy { Handler(Looper.getMainLooper()) }
+    override fun imageUrlParse(response: Response): String = ""
 
-    private fun showToast(message: String) {
-        handler.post {
-            Toast.makeText(application, message, Toast.LENGTH_LONG).show()
-        }
+    // ============================== Utilities =================================
+
+    private fun tryCompatibility(manga: SManga): SManga {
+        val substring = mangaSubStrings.first { manga.url.contains(it, ignoreCase = true) }
+        val slug = manga.url
+            .substringAfterLast("$substring/")
+            .substringBeforeLast("/")
+        return database.values.firstOrNull { it.slug == slug }
+            ?.toSManga()
+            ?: throw IOException("Migrate a obra para a $name")
     }
```

Comment:
Your code suggests the URL format has changed to using an ID. You should bump the `versionId` to force users to migrate.

Reply:
It's also preferred to clear the old preferences:

https://github.com/keiyoushi/extensions-source/blob/572201d819933e0ac48f5e87506d87ecd9617bea/src/zh/sixmh/src/eu/kanade/tachiyomi/extension/zh/sixmh/SixMH.kt#L21-L26